core/iwasm/interpreter/wasm_loader.c

```diff
@@ -3875,6 +3875,13 @@ create_module(char *error_buf, uint32 error_buf_size)
     }
 #endif
 
+#if WASM_ENABLE_SHARED_MEMORY != 0
+    if (os_mutex_init(&module->mem_lock) != 0) {
+        set_error_buf(error_buf, error_buf_size, "init memory lock failed");
+        return NULL;
```

Comment:
Should destroy resources before `return NULL`:
```C
#if WASM_ENABLE_DEBUG_INTERP != 0                    \
    || (WASM_ENABLE_FAST_JIT != 0 && WASM_ENABLE_JIT != 0 \
        && WASM_ENABLE_LAZY_JIT != 0)
        os_mutex_destroy(&module->instance_list_lock);
#endif
        wasm_runtime_free(module);
```

Reply:
And found that L3868 has typo, could you help to modify:
`WASM_ENABLE_FAST_JIT != 0 && WASM_ENABLE_JIT`
to
`WASM_ENABLE_FAST_JIT != 0 && WASM_ENABLE_JIT != 0`